site/src/models/nfts.js

```diff
@@ -109,32 +81,37 @@ export async function list(prefix, options) {
     throw new Error('invalid filter')
   }
 
-  let done = false
-  let cursor
-  /** @type Promise<KVValue<NFT>>[] */
-  const nfts = []
-
-  while (!done) {
-    // @ts-ignore
-    const nftIdxList = await stores.nftsIndex.list({ prefix, cursor })
-
-    for (const k of nftIdxList.keys) {
-      const { created } = decodeIndexKey(k.name)
-      if (created.getTime() < before.getTime()) {
-        // @ts-ignore
-        nfts.push(stores.nfts.get(k.metadata.key, 'json'))
-        if (nfts.length >= limit) {
-          break
-        }
+  /** @type {Array<{ nftPromise: Promise<NFT|null>, indexData: import('./nfts-index.js').IndexData }>} */
+  const items = []
+  for await (const [key, data] of nftsIndex.entries(prefix)) {
+    if (new Date(key.created).getTime() < before.getTime()) {
+      items.push({
+        nftPromise: stores.nfts.get(data.key, 'json'),
+        indexData: data,
+      })
+      if (items.length >= limit) {
+        break
       }
     }
-
-    if (nfts.length >= limit) {
-      break
-    }
-    cursor = nftIdxList.cursor
-    done = nftIdxList.list_complete
   }
 
-  return Promise.all(nfts)
+  return Promise.all(
+    items.map(async ({ nftPromise, indexData }) => {
+      const nft = await nftPromise
+      return nft
+        ? {
+            ...nft,
+            size: indexData.size,
+            pin: {
+              cid: nft.cid,
+              ...(nft.pin || {}),
+              status: indexData.pinStatus,
+              size: indexData.size,
+              created: nft.created,
+            },
+            deals: [], // FIXME: how to get deals data?
```

Comment:
we cant get from the deals KV ?

Reply:
Yes we could, but we risk hitting API limits...this endpoint currently does not return this information so I was punting the solution until I had a better idea!

Reply:
KV calls limit ?

Reply:
Seems to be undocumented but definitely exists.